pilz_trajectory_generation/include/pilz_trajectory_generation/trajectory_generation_exceptions.h

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2019 Pilz GmbH & Co. KG
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef TRAJECTORY_GENERATION_EXCEPTIONS_H
+#define TRAJECTORY_GENERATION_EXCEPTIONS_H
+
+#include <stdexcept>
+#include <string>
+
+#include <moveit_msgs/MoveItErrorCodes.h>
+
+namespace pilz_trajectory_generation
+{
+
+/**
+ * @brief Exception storing an moveit_msgs::MoveItErrorCodes value.
+ */
+class MoveItErrorCodeException : public std::runtime_error
+{
+public:
+  MoveItErrorCodeException(const std::string& msg);
+
+protected:
+  MoveItErrorCodeException(const MoveItErrorCodeException&) = default;
+  MoveItErrorCodeException(MoveItErrorCodeException&&) = default;
+  ~MoveItErrorCodeException() = default;
+
+  MoveItErrorCodeException& operator=(const MoveItErrorCodeException&) = default;
+  MoveItErrorCodeException& operator=(MoveItErrorCodeException&&) = default;
+
+public:
+  virtual const moveit_msgs::MoveItErrorCodes::_val_type& getErrorCode() const = 0;
+};
+
+template <moveit_msgs::MoveItErrorCodes::_val_type ERROR_CODE = moveit_msgs::MoveItErrorCodes::FAILURE>
+class TemplatedMoveItErrorCodeException : public MoveItErrorCodeException
+{
+public:
+  TemplatedMoveItErrorCodeException(const std::string& msg);
+  TemplatedMoveItErrorCodeException(const std::string& msg,
+                                    const moveit_msgs::MoveItErrorCodes::_val_type& error_code);
```

Comment:
The PlanningPipelineException needs the error code to be set at construction time. This is needed to be compatible with old code. Otherwise, I would have to refactor ALL old code. => Done.